Description: fix hash_permutation on big endian
Author: moneromooo-monero <moneromooo-monero@users.noreply.github.com>
Origin: upstream, https://github.com/monero/monero-core/commit/db24a2e
Last-Update: 2018-11-20
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/crypto/hash.c
+++ b/src/crypto/hash.c
@@ -36,7 +36,14 @@
 #include "keccak.h"
 
 void hash_permutation(union hash_state *state) {
+#if BYTE_ORDER == LITTLE_ENDIAN
   keccakf((uint64_t*)state, 24);
+#else
+  uint64_t le_state[25];
+  memcpy_swap64le(le_state, state, 25);
+  keccakf(le_state, 24);
+  memcpy_swap64le(state, le_state, 25);
+#endif
 }
 
 void hash_process(union hash_state *state, const uint8_t *buf, size_t count) {
